### PR TITLE
feat(filter-editor)!: replace column names by column types

### PR DESCRIPTION
**Pull request by ninofiliu**
This is a breaking change
Associated Tucana PR: https://github.com/ToucanToco/tucana/pull/8771

The filter editor used to create an all-string condition, but now it takes advantage of the column types to build a typed condition

- before
  - column names prop
    ```json
    ["name", "age"]
    ```
  - condition
    ```json
    {
      "and": [
        { "column": "name", "operator": "eq", "value": "john" },
        { "column": "age", "operator": "eq", "value": "10" }
      ]
    }
    ```
- after
  - column types prop
    ```json
    { "name": "string", "age": "integer" }
    ```
  - condition (note the age value is of the right type)
    ```json
    {
      "and": [
        { "column": "name", "operator": "eq", "value": "john" },
        { "column": "age", "operator": "eq", "value": 10 }
      ]
    }
    ```

### CODE REVIEW

File: src/components/FilterEditor.vue [hunk 1]
```diff
@@ -67,7 +69,7 @@ export default class FilterEditor extends Vue {
   errors!: ErrorObject[];
 
   get conditionsTree() {
-    return buildConditionsEditorTree(this.filterTree);
+    return buildConditionsEditorTree(castFilterStepTreeValue(this.filterTree, this.columnTypes));
   }
```
Comment (davinov) [inline src/components/FilterEditor.vue:73]: Now that this is done in the FilterEditor level, it should be removed from FilterStepForm.vue as well

Comment (davinov) [reply src/components/FilterEditor.vue:73]: Btw, I think it's also used in the IFThenElseStepForm

**Review by davinov — CHANGES_REQUESTED** [review]
Seems great! Here a just a few cleaning comments:

---
Also, in order to complete the weaverbird's release, you'll need to:
- update the changelog
- update the version sonar-properties files
and when it's merged in master, just create a release in GitHub and an action will publish it to npm. :tada:

**Comment by WilliamGorge** [comment]
merci @ninofiliu !

dans la description je crois que tu t'es trompé dans le `after`, je pense que voulais écrire
```
    { "column": "age", "operator": "eq", "value": 10 }
```
non?

**Comment by ninofiliu** [comment]
Ah oui en effet! Desc corrigée

File: src/components/FilterEditor.vue (hunk 1 above)
Comment (ninofiliu) [reply src/components/FilterEditor.vue:73]: https://github.com/ToucanToco/weaverbird/pull/767/commits/2446543ebee2dc7a38204b3266a406b81e148104 https://github.com/ToucanToco/weaverbird/pull/767/commits/7dab6fbec53680c897c0804bc63be9dd075b73d6 https://github.com/ToucanToco/weaverbird/pull/767/commits/5789eae618450027d37ab79c52e9542908390277 https://github.com/ToucanToco/weaverbird/pull/767/commits/4e0a4e1d152291cfeb844a4800c47f5e44a656c5

**Comment by ninofiliu** [comment]
@davinov par contre c'est breaking donc on devrait garder le point d'exclamation dans le nom de pr non?

> a commit that has a [...] ! after the type/scope, introduces a breaking API change (correlating with MAJOR in semantic versioning)

from https://www.conventionalcommits.org/en/v1.0.0/#summary

**Comment by davinov** [comment]
> @davinov par contre c'est breaking donc on devrait garder le point d'exclamation dans le nom de pr non?
> 
> > a commit that has a [...] ! after the type/scope, introduces a breaking API change (correlating with MAJOR in semantic versioning)
> 
> from https://www.conventionalcommits.org/en/v1.0.0/#summary

Ah c'est ca que ca signifie ! je croyais que c'était une typo ^^ sorry !

**Review by davinov — APPROVED** [review]
Thx!

**Pull request merged**